features/seasonTeams/restRequestHandlers/createSeasonTeamRequestHandler.ts

```diff
@@ -0,0 +1,52 @@
+import { seasonIdParameterValidator } from "@/features/common/paramValidators/seasonIdParameterValidator";
+import { SeasonIdParams } from "@/features/common/params/seasonIdParams";
+import { prismaClient } from "@/features/common/prisma/prismaClient";
+import type { RestRequestHandlerBuilderOptions } from "@/features/common/restRequestHandlers/restRequestHandlerBuilder";
+import { restRequestHandlerBuilder } from "@/features/common/restRequestHandlers/restRequestHandlerBuilder";
+import { seasonTeamSelector } from "@/features/seasonTeams/prismaSelectors/seasonTeamSelector";
+import { seasonTeamDetailsResponse } from "@/features/seasonTeams/restResponses/seasonTeamDetailsResponse";
+import { SeasonTeamCreateInputObjectSchema } from "@/prisma/generated/schemas/objects/SeasonTeamCreateInput.schema";
+import type { Prisma } from "@prisma/client";
+import type { NextRequest } from "next/server";
+
+const createSeasonTeamRequestHandlerBuilderOptions: RestRequestHandlerBuilderOptions<SeasonIdParams, Prisma.SeasonTeamCreateInput> = {
+    onValidateParams: seasonIdParameterValidator,
+    onValidateRequestAsync: async (req: NextRequest) => {
+        const requestBody = await req.json();
+        const validation = SeasonTeamCreateInputObjectSchema.safeParse(requestBody);
+        if (!validation.success) {
+            const { errors } = validation.error;
+            return { success: false, issues: errors };
+        }
+        return { success: true, validatedRequestBody: validation.data };
+    },
+    onValidRequestAsync: async (req: NextRequest, details) => {
+        if (details && details.params && details.validatedRequestBody) {
+            const createArgs: Prisma.SeasonTeamCreateArgs = {
+                data: details.validatedRequestBody,
+                select: seasonTeamSelector
+            };
+            const seasonTeam = await prismaClient.seasonTeam.create(createArgs);
+            const { params } = details.params;
+            const { seasonId } = params;
+            const seasonUpdateArgs: Prisma.SeasonUpdateArgs = {
+                where: {
+                    id: seasonId
+                }, data: {
+                    teams: {
+                        create: [details.validatedRequestBody]
+                    }
+                },
+                select: {
+                    teams: { select: seasonTeamSelector }
+                }
+            }
+            await prismaClient.season.update(seasonUpdateArgs)
```

Comment:
Bon, ici, c'est bête, mais je n'ai pas trouvé comment le faire en une seule transaction, et avoir un retour de la transation qui explicitement retournait les seasonTeam bien "typable". 

Étrangement, la doc de prisma était beaucoup plus claire pour le cas où on voulait créer le parent considérant un `nested` existant, ou alors comment faire le parent et un ou plusieurs "nested" dans la même transation, mais créer le nested et `connect` au parent, sans le créer s'il n'existe pas, ça semblait moins commode.

Je suis peut-être passé à côté de quelque chose, mais bon, pour l'instant ça ça devrait faire. Si tu vois comment je suis preneuse.

Reply:
Je pense que tu as fait exactement une transaction déjà. Si tu lis [la doc ici](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#nested-writes-1)
: "If any operation fails, Prisma Client rolls back the entire transaction."